File: src/leetcode/2026/01/Solution507.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool checkPerfectNumber(int num) {
        if (num <= 1) {
            return false;
        }
        int sum = 1; // 1 is a proper divisor
        for (int i = 2; i * i <= num; ++i) {
            if (num % i == 0) {
                sum += i;
                if (i != num / i) {
                    sum += num / i;
                }
            }
        }
        return sum == num;
    }
};

int main() {
    Solution solution;

    int num1 = 28;
    cout << (solution.checkPerfectNumber(num1) ? "true" : "false") << endl; // Expected output: true

    int num2 = 7;
    cout << (solution.checkPerfectNumber(num2) ? "true" : "false") << endl; // Expected output: false

    return 0;
}

File: src/leetcode/2026/01/Solution507.java
public class Solution507 {
    public boolean checkPerfectNumber(int num) {
        if (num <= 1) {
            return false;
        }
        int sum = 1; // 1 is a proper divisor
        int sqrtNum = (int) Math.sqrt(num);
        for (int i = 2; i <= sqrtNum; i++) {
            if (num % i == 0) {
                sum += i;
                int otherDivisor = num / i;
                if (otherDivisor != i) {
                    sum += otherDivisor;
                }
            }
        }
        return sum == num;
    }

    public static void main(String[] args) {
        Solution507 solution = new Solution507();

        int num1 = 28;
        System.out.println("Is " + num1 + " a perfect number? " + solution.checkPerfectNumber(num1));

        int num2 = 12;
        System.out.println("Is " + num2 + " a perfect number? " + solution.checkPerfectNumber(num2));
    }
}